scene/main/window.cpp

```diff
@@ -389,6 +389,12 @@ void Window::move_to_center() {
 }
 
 void Window::set_size(const Size2i &p_size) {
+#if defined(ANDROID_ENABLED)
+	if (!get_parent()) {
+		// Can't set root window size on Android.
+		return;
+	}
+#endif
 	ERR_MAIN_THREAD_GUARD;
```

Comment:
The thread guard check should still happen before accessing the scene tree with `get_parent`.